> > I cannot modify the pandas/_libs/parsers.pyx without running into issues with cythonize (via pip install -e .).
> 
> What issues do you see? You'll need a C compiler, since you're updating a Cython file. Typically I use `python setup.py build_ext --inplace`, but `pip install -e .` may do the same.

I was adding an `if lzma is None` to one of the lines to see if `RuntimeError` is necessary.
However, when I run `pip install -e .` after modifying the file, I get:
```
    ERROR: Complete output from command python setup.py egg_info:
    ERROR: 
    Error compiling Cython file:
    ------------------------------------------------------------
    ...
    
    
    cdef extern from "parser/tokenizer.h":
    
        ctypedef enum ParserState:
    	START_RECORD
    ^
    ------------------------------------------------------------
    
    pandas/_libs/parsers.pyx:90:0: Mixed use of tabs and spaces
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/Users/guilhermesalome/Desktop/pandas/setup.py", line 813, in <module>
        ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
      File "/Users/guilhermesalome/Desktop/pandas/setup.py", line 542, in maybe_cythonize
        return cythonize(extensions, *args, **kwargs)
      File "/Users/guilhermesalome/.pyenv/versions/miniconda3-latest/envs/pandas-dev/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1096, in cythonize
        cythonize_one(*args)
      File "/Users/guilhermesalome/.pyenv/versions/miniconda3-latest/envs/pandas-dev/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1219, in cythonize_one
        raise CompileError(None, pyx_file)
    Cython.Compiler.Errors.CompileError: pandas/_libs/parsers.pyx
    Compiling pandas/_libs/parsers.pyx because it changed.
    [1/1] Cythonizing pandas/_libs/parsers.pyx
    ----------------------------------------
ERROR: Command "python setup.py egg_info" failed with error code 1 in /Users/guilhermesalome/Desktop/pandas/
```
I am modifying the file in Emacs without any mode for Cython.

--- a/doc/source/whatsnew/v0.25.1.rst
+++ b/doc/source/whatsnew/v0.25.1.rst
@@ -103,7 +103,7 @@ MultiIndex
 
 I/O
 ^^^
-
+- Bug: importing `lzma` can fail when Python is not properly installed. For example, on MacOS, installing Python with `pyenv` when `xz` is not available does not raise any issues during installation. However, trying to `import pandas` afterwards will fail, since the module `lzma` will not be available. Usually, this can be solved by installing the appropriate Python dependencies, and then resintalling Python. This fix warns the user that Python was not properly installed during a `import pandas`. In this case, Pandas can still be used (not possible before this fix), but any call to the `lzma` module will raise a `RuntimeError`. (:issue: `27575`)
 -
 -
 -

--- a/pandas/_libs/parsers.pyx
+++ b/pandas/_libs/parsers.pyx
@@ -646,6 +646,7 @@ cdef class TextReader:
                     raise ValueError('Multiple files found in compressed '
                                      'zip file %s', str(zip_names))
             elif self.compression == 'xz':
+                if lzma is None: raise RuntimeError("lzma module not available.")
                 if isinstance(source, str):
                     source = lzma.LZMAFile(source, 'rb')
                 else:

--- a/pandas/compat/__init__.py
+++ b/pandas/compat/__init__.py
@@ -10,6 +10,8 @@ Other items:
 import platform
 import struct
 import sys
+import warnings
+
 
 PY35 = sys.version_info[:2] == (3, 5)
 PY36 = sys.version_info >= (3, 6)
@@ -68,8 +70,6 @@ def is_platform_32bit():
 
 
 def import_lzma():
-    import warnings
-
     try:
         import lzma
 

